### DISCUSSION TITLE
Built-in instances for basic scala types

### DISCUSSION THREAD

**Discussion by amricko0b** [andyglow/scala-jsonschema#282] [General]
Greetings!

I've got a function this little function (implementation omitted, cause I think it doesn't matter)
```
def doSomething[T: Schema](t: T) = // Does something with T and Schema[T]
``` 

When I try to compile, for example, `doSomething[Int]` or `doSomething[Map[String, Int]]` - compilation fails due to lack of implicit for `Schema[Int]` or `Schema[Map[String, Int]]` respectively.

The only workaround I see for now is to derive all instances manually, but I suppose the library already has them somewhere, due to support out-of-the-box.

Is there a way to somehow import instances for scala types (for example, I need instance for `Map`) rather then derive them?

Thanks for attention!

**Comment by andyglow** [comment]
Hello, @amricko0b 
you can try few things, see what feets best

*Option 1*
```
import json.Json.auto_

def doSomething[T: Schema](t: T) = ???

doSomething[Int](55)
doSomething[Map[String, Int]](Map.empty)
```

*Option 2*
```
def doSomething[T](t: T)(implicit schema: Schema[T] = json.Json.schema[T]) = ???

doSomething[Int](55)
doSomething[Map[String, Int]](Map.empty)
```

**Reply by amricko0b** [reply]
Hello, @andyglow ! Thanks for your reply.

I've tried both options, but they don't compile (may be I've done something wrong).
Here I provide some detailed examples, to make my question more substantive. I try to derive a schema for a type and then convert it to Circe json AST.

Option 1:
```
import munit.FunSuite
import com.github.andyglow.jsonschema.AsCirce._
import json.Json.auto._
import json.Schema
import json.schema.Version.Draft07
import io.circe.Json

class JsonSchemaSuite extends FunSuite {

  def createCirceSchema[T](implicit schema: Schema[T]): Json =
    schema.asCirce(Draft07("id"))

  test("Derived schema as Circe json value") {

    println(createCirceSchema[Int])

  }
}
```

Compilation fails with:  `could not find implicit value for parameter schema: json.Schema[Int]`

Option 2:
```
import munit.FunSuite
import com.github.andyglow.jsonschema.AsCirce._
import json.{Json => JsonSchema}
import json.Schema
import json.schema.Version.Draft07
import io.circe.Json

class JsonSchemaSuite extends FunSuite {

  def createCirceSchema[T](implicit schema: Schema[T] = JsonSchema.schema[T]): Json =
    schema.asCirce(Draft07("id"))

  test("Derived schema as Circe json value") {

    println(createCirceSchema[Int])

  }
}

```

this ends up with: `schema for T is not supported`.

I use scala 2.13.8 to compile.
